### PR TITLE
Refactored Blocks/ToC/Edits.jsx to functional component

### PR DESCRIPTION
**Pull request by Prince0906**
Ref: #4460 
### **About This PR**
This PR refactored 

> packages/volto/src/components/manage/Blocks/ToC/Edit.jsx

 from class based to functional component

**Before:**

https://github.com/user-attachments/assets/84673f37-1e76-4382-a247-02ac82645ea4

**After**

https://github.com/user-attachments/assets/b007d68f-960b-4105-b9b1-682136457548

This is my first time contrubuting to Volto .So,if I have done anything wrong in my PR, please give feedback and I will try to rectify that.

### CODE REVIEW

**Comment by Prince0906** [comment]
@stevepiercy Can you review my pr, it's ready to preview.

File: packages/volto/news/6167.feature
Comment (stevepiercy) [inline packages/volto/news/6167.feature:1]:
```suggestion
Refactor the table of contents block component from a class component to a functional component. @Prince0906
```

**Review by stevepiercy — CHANGES_REQUESTED** [review]
This needs review from a member of the @plone/volto-team.

I noticed several recent pull requests that provide similar refactors from newcomers. I'm curious whether this PR is part of a GSoC project or some other recruitment or discovery?

**Review by ichim-david — APPROVED** [review]

**Review by stevepiercy — APPROVED** [review]
LGTM.

**Comment by Prince0906** [comment]
> This needs review from a member of the @plone/volto-team.
> 
> I noticed several recent pull requests that provide similar refactors from newcomers. I'm curious whether this PR is part of a GSoC project or some other recruitment or discovery?

I am just exploring this things with my friends.

**Comment by Prince0906** [comment]
@sneridagh please review it

**Comment by Prince0906** [comment]
@sneridagh please review my pr

**Comment by stevepiercy** [comment]
@Prince0906 you do not need to repeatedly `@` individual people. It is considered annoying and rude. I have already tagged the appropriate team.

**Comment by ichim-david** [comment]
@Prince0906 Merging a pull request takes time and you need to be patient. There are some extra things to take into consideration when something is merged such as the need to mark something as breaking or not, or to think on what release version this change should be added.

The good news is that you have 2 approved reviews so this work will be merged however asking sneridan to review it
will not get your pull request merged quickly, instead you will get a notice about not repeatably messaging someone.

I know that you are anxious to have this merged  but again that takes time. While waiting if you see another issue that you would know how to tackle feel free to do so and we can help then just like we did now.

**Comment by Prince0906** [comment]
> @Prince0906 Merging a pull request takes time and you need to be patient. There are some extra things to take into consideration when something is merged such as the need to mark something as breaking or not, or to think on what release version this change should be added.
> 
> The good news is that you have 2 approved reviews so this work will be merged however asking sneridan to review it will not get your pull request merged quickly, instead you will get a notice about not repeatably messaging someone.
> 
> I know that you are anxious to have this merged but again that takes time. While waiting if you see another issue that you would know how to tackle feel free to do so and we can help then just like we did now.

Sorry for the mistake, I will be patient and will avoid making this mistake again.

**Comment by stevepiercy** [comment]
I went through the original PLIP and found that this PR may duplicate work in https://github.com/plone/volto/pull/4961.

**Comment by Prince0906** [comment]
> I went through the original PLIP and found that this PR may duplicate work in #4961.

**Comment by Prince0906** [comment]
> I went through the original PLIP and found that this PR may duplicate work in #4961.

In that pr , Because of some other fixing all checks are not passing.

**Comment by stevepiercy** [comment]
@Prince0906 OK. Thanks for confirming. I've put this (and dozens of other) PRs on our next Volto Team Meeting agenda. @plone/volto-team needs to do better with its management of PRs.

**Comment by Prince0906** [comment]
@stevepiercy I had updated the branch and this failed check is not related to my component.

**Comment by stevepiercy** [comment]
@Prince0906 thanks for the update and confirming your results against the CI failing check. I restarted the [failing check](https://github.com/plone/volto/actions/runs/10124451422/job/27998815052?pr=6167) as it has a flaky test in it (metadata).

**Comment by stevepiercy** [comment]
Flaky test passed. We need a final review before merging from @plone/volto-team to see if it is a breaking change, and compare against #4961.

File: packages/volto/src/components/manage/Blocks/ToC/Edit.jsx
Comment (davisagli) [inline packages/volto/src/components/manage/Blocks/ToC/Edit.jsx:10]:
```suggestion
  const {
      onChangeBlock,
      data,
      block,
      selected,
      navRoot,
      contentType,
      blocksErrors,
  } = props;
```

File: packages/volto/src/components/manage/Blocks/ToC/Edit.jsx
Comment (davisagli) [inline packages/volto/src/components/manage/Blocks/ToC/Edit.jsx:17]:
```suggestion
    onChangeBlock,
    data,
    block,
    selected,
    navRoot,
    contentType,
    blocksErrors,
```

**Review by davisagli — APPROVED** [review]
Thanks for your contribution!

I'll go ahead and merge this, and take a separate look at https://github.com/plone/volto/pull/4961, which also includes a fix that is unrelated to the refactoring.

**Pull request merged**
